junit-vintage-engine/src/main/java/org/junit/vintage/engine/discovery/IsPotentialJUnit4TestClass.java

```diff
@@ -22,12 +23,20 @@
 
 	@Override
 	public boolean test(Class<?> candidate) {
-		// Do not collapse into single return.
 		if (isAbstract(candidate))
 			return false;
 		if (!isPublic(candidate))
 			return false;
-		return !candidate.isMemberClass();
+		// Do not collapse into single return.
```

Comment:
Well, before the warning was on the second `if`. My point is: You could write this method body as

```java
return !isAbstract(candidate) && isPublic(candidate) && !isNonStaticMemberClass(candidate);
```

but we prefer the `if` statements in this case. Thus, the comment is valid for the whole method body and therefore should be at the very top. Does that make sense?